### PR TITLE
Create CPPModernization.rst

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Outcome from code camp. This is a proposal for a suggested list of changes to apply as part of applying the "Boy Scout Rule".

Fixes https://github.com/mantidproject/mantid/issues/25459.

### CODE REVIEW

**Review by martyngigg — DISMISSED** [review]
This is a good list of points to form a base of guidelines going forward.

**Comment by DanNixon** [comment]
Does this need referenced from one of the other docs pages? (e.g. [Standards](http://developer.mantidproject.org/Standards/index.html))

File: dev-docs/source/Standards/CPPModernization.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+=================================
+
+Whenever editing code, try to follow the "Boy Scout Rule", i.e., leave code surrounding your in a better state than before.
+For Mantid in particular, pay attention to the following:
+
+- Do not pass ``shared_ptr`` unless you are passing ownership. Just pass ``const MyClass &`` or ``MyClass &`` instead.
+- Replace uses of ``new`` by ``std::make_unique``, where possible.
+- Split large functions/methods into smaller functions/methods. Consider using free functions since that facilitates unit testing and code reuse.
+- Remove trivial or obvious comments. If a comment can be changed into a better variable name, do so.
+- Where it makes sense, replace boolean or integer flags by enum or preferably enum class. Even better: Avoid them entirely, e.g., by moving code into two separate free functions.
```
Comment (peterfpeterson) [inline dev-docs/source/Standards/CPPModernization.rst]: Minor mark-up change:
```suggestion
- Where it makes sense, replace boolean or integer flags by ``enum`` or preferably ``enum class``. Even better: Avoid them entirely, e.g., by moving code into two separate free functions.
```

**Review by peterfpeterson — APPROVED** [review]
:shipit:

**Pull request merged**